osbuild/sources.py

```diff
@@ -51,17 +54,66 @@ class SourceService(host.Service):
     """Source host service"""
 
     @abc.abstractmethod
-    def download(self, items, cache, options):
-        pass
+    def fetch_one(self, checksum, desc) -> None:
+        """Performs the actual fetch of an element described by its checksum and its descriptor"""
+
+    def config_download(self) -> None:
+        """Hook to set config for the download affecting all the elements, such as parallelism or special caches."""
```

Comment:
Seeing how this is used later, where it sets instance global variables, we should rename this to `setup` can then call this remotely like `download` before the latter is called.